examples/rdkafka_performance.c

```diff
@@ -1400,11 +1400,16 @@ int main (int argc, char **argv) {
 			cnt.msgs++;
 			cnt.bytes += msgsize;
 
-                        if (rate_sleep)
-                                do_sleep(rate_sleep);
-
 			/* Must poll to handle delivery reports */
-			rd_kafka_poll(rk, 0);
+			if (rate_sleep) {
+				rd_ts_t next = rd_clock() + (rd_ts_t) rate_sleep;
+				do {
+					int poll_timeout = RD_MAX(0, (next - rd_clock()) / 1000);
```

Comment:
this variable is only used once, move the right side directly into poll() argument

Reply:
need int cast as so:
`(int)((next - rd_clock()) / 1000)`

Reply:
Done, and re-indented to avoid a long line.